pkg/fuzzy/google_translate.go

```diff
@@ -230,3 +195,250 @@ func textToBatches(text []string, batchLimit int) [][]string { //nolint:unused
 
 	return batches
 }
+
+// getTexts extracts translatable text from the translation.Messages slice.
+// To prevent the translation of placeholders, they are replaced with simplified
+// numbered placeholders '{$d}', where the placeholder number represents the
+// index of the ast.PlaceholderPattern element in the message AST.
+// The function returns a slice of strings representing the extracted translatable texts and error.
+//
+// Example:
+// Input:
+// 	&model.Translation{
+// 		Original: false,
+// 		Language: language.English,
+// 		Messages: []model.Message{
+// 			{
+// 				ID:      "Hello World!",
+// 				Message: `Hello, { |name| :function } { |lastName| :function2 }!`,
+// 				Status:  model.MessageStatusUntranslated,
+// 			},
+// 		},
+// 	}
+
+// Output:
+// []string{"Hello, {$0} {$1}!"}, nil .
+func getTexts(translation *model.Translation) ([]string, error) {
+	texts := make([]string, 0, len(translation.Messages))
+
+	for i := range translation.Messages {
+		messageAST, err := ast.Parse(translation.Messages[i].Message)
+		if err != nil {
+			return nil, fmt.Errorf("parse mf2 message with ID '%s': %w", translation.Messages[i].ID, err)
+		}
+
+		switch v := messageAST.Message.(type) {
+		default:
+			return nil, fmt.Errorf("unsupported message type: %T", v)
+		case ast.SimpleMessage:
+			texts = append(texts, printPattern(v.Patterns))
+		case ast.ComplexMessage:
+			switch v := v.ComplexBody.(type) {
+			case ast.Matcher:
+				for _, variant := range v.Variants {
+					texts = append(texts, printPattern(variant.QuotedPattern.Patterns))
+				}
+			case ast.QuotedPattern:
+				texts = append(texts, printPattern(v.Patterns))
+			}
+		}
+	}
+
+	return texts, nil
+}
+
+// printPattern iterates over an ast.Pattern slice, appending ast.TextPatterns to a string.
+// When an ast.PlaceholderPattern is encountered, a simplified placeholder version '{$d}' is appended.
+// The function returns a string representing the concatenated patterns.
+// Example:
+// Input:
+//
+//	[]ast.Patterns{
+//		TextPattern("Hello"),
+//		PlaceholderPattern{ Expression: LiteralExpression{Literal: QuotedLiteral("name")}},
+//		TextPattern(" "),
+//		PlaceholderPattern{ Expression: LiteralExpression{Literal: QuotedLiteral("lastName")}},
+//		TextPattern("!"),
+//	}
+//
+// Output:
+// "Hello {$0} {$1}!".
+func printPattern(pattern []ast.Pattern) string {
```

Comment:
**ambiguos**
The term “print” is commonly used in to refer to outputting information to the console. This function doesn’t print anything; it returns a string.

Reply:
We have func with similar naming in the go-mf2 pkg.

Reply:
dont think this is good :)